File: List/print_reverse.py
class node:
    def __init__(self, data):
        self.data = data
        self.next = None


class ListNode:

    def __init__(self):
        self.head = None

    def insert(self, val):
        if self.head is None:
            self.head = node(val)
        else:
            temp = self.head

            while temp.next is not None:
                temp = temp.next

            temp.next = node(val)

    def display(self):
        temp = self.head

        while temp is not None:
            print(temp.data)
            temp = temp.next


    def print_reverse(self):
        stack = []

        tmp = self.head

        while tmp is not None:
            stack.append(tmp)
            tmp = tmp.next
        
        while len(stack) > 0:
            print(stack.pop().data)



if __name__ == "__main__":

    l = ListNode()

    for i in range(5):
        i+=1
        l.insert(i)

    l.print_reverse()
    

File: List/print_reverse.cpp
#include<bits/stdc++.h>
using namespace std;

struct node{
    int data;
    struct node *next;
};

typedef struct node *N;

N getnode(int x){
    N t=(struct node *)malloc(sizeof(struct node));
    t->data = x;
    t->next = NULL;
    return t;
}

N insert(N start,int x){
    N t = getnode(x);

    if(!start){
        return t;
    }

    N c = start;

    while(c->next!=NULL){
        c=c->next;
    }

    c->next = t;
    return start;
}

void display(N start){
    N t=start;

    while(t){
        cout << t->data << endl;
        t=t->next;
    }

}

void print_reverse(N start){
    stack<N>stk;

    N t = start;
    while(t){
        stk.push(t);
        t=t->next;
    } 

    while(!stk.empty()){
        cout << stk.top()->data << endl;
        stk.pop();
    }
}

int main(){
    N start = NULL;

    for(int i=1;i<=5;i++){
        start = insert(start, i);
    }

    cout << "Print reverse of SLL:- " << endl;
    
    print_reverse(start);

    return 0;
}



/* This code is contributed by Vittal AB*/